sabnzbd/assembler.py

```diff
@@ -191,7 +188,10 @@ def assemble(nzo: NzbObject, nzf: NzbFile, file_done: bool):
                     # Could be empty in case nzo was deleted
                     if data:
                         fout.write(data)
-                        nzf.md5.update(data)
+                        if len(data) == nzo.article_size:
+                            nzf.crc32 = crc_multiply(nzf.crc32, nzo.crc32_coeff) ^ article.crc32
+                        else:
+                            nzf.crc32 = crc_concat(nzf.crc32, article.crc32, len(data))
```

Comment:
Maybe we can have an `nzf.update_crc32(data, article.crc)` function instead of doing this here?